src/test/java/software/aws/neptune/opencypher/OpenCypherIAMRequestGeneratorTest.java

```diff
@@ -0,0 +1,157 @@
+
+import com.amazonaws.http.HttpMethodName;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.neo4j.driver.AuthToken;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.internal.security.InternalAuthToken;
+
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.amazonaws.auth.internal.SignerConstants.AUTHORIZATION;
+import static com.amazonaws.auth.internal.SignerConstants.HOST;
+import static com.amazonaws.auth.internal.SignerConstants.X_AMZ_DATE;
+import static com.amazonaws.auth.internal.SignerConstants.X_AMZ_SECURITY_TOKEN;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.neo4j.driver.Values.value;
+import static org.neo4j.driver.internal.security.InternalAuthToken.CREDENTIALS_KEY;
+import static org.neo4j.driver.internal.security.InternalAuthToken.PRINCIPAL_KEY;
+import static org.neo4j.driver.internal.security.InternalAuthToken.REALM_KEY;
+import static org.neo4j.driver.internal.security.InternalAuthToken.SCHEME_KEY;
+import static software.aws.neptune.opencypher.OpenCypherIAMRequestGenerator.DUMMY_USERNAME;
+import static software.aws.neptune.opencypher.OpenCypherIAMRequestGenerator.HTTP_METHOD_HDR;
+import static software.aws.neptune.opencypher.OpenCypherIAMRequestGenerator.SERVICE_NAME;
+
+class OpenCypherIAMRequestGeneratorTest {
+    private static final String DUMMY_ACCESS_KEY_ID = "xxx";
+    private static final String DUMMY_SESSION_TOKEN = "zzz";
+
+    @BeforeAll
+    public static void beforeAll() {
+        System.setProperty("aws.accessKeyId", DUMMY_ACCESS_KEY_ID);
+        System.setProperty("aws.secretKey", "yyy");
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        System.clearProperty("aws.accessKeyId");
+        System.clearProperty("aws.secretKey");
+    }
+
+    @AfterEach
+    public void afterEach() {
+        System.clearProperty("aws.sessionToken");
+    }
+
+    @Test
+    public void testCreateAuthToken() {
+        verifyAuthToken(false);
+    }
+
+    @Test
+    public void testCreateAuthTokenWithTempCreds() {
+        System.setProperty("aws.sessionToken", DUMMY_SESSION_TOKEN);
+        verifyAuthToken(true);
+    }
+
+    private void verifyAuthToken(final boolean useTempCreds) {
+        final String url = "bolt://somehost.com:58763";
```

Comment:
Is the default protcol `bolt` only ? I remember we had something like bolt+ssl ?

Reply:
It doesn't matter for the signature. The unit test works regardless of the protocol. It might matter for connectivity, but that's not in the scope of this PR and not changed.